store/rootmulti/proof.go

```diff
@@ -37,99 +16,12 @@ func RequireProof(subpath string) bool {
-
-// the multi-store proof operation constant value
-const ProofOpMultiStore = "multistore"
-
-// TODO: document
-type MultiStoreProofOp struct {
-	// Encoded in ProofOp.Key
-	key []byte
-
-	// To encode in ProofOp.Data.
-	Proof *MultiStoreProof `json:"proof"`
-}
-
-func NewMultiStoreProofOp(key []byte, proof *MultiStoreProof) MultiStoreProofOp {
-	return MultiStoreProofOp{
-		key:   key,
-		Proof: proof,
-	}
-}
-
-// MultiStoreProofOpDecoder returns a multi-store merkle proof operator from a
-// given proof operation.
-func MultiStoreProofOpDecoder(pop merkle.ProofOp) (merkle.ProofOperator, error) {
-	if pop.Type != ProofOpMultiStore {
-		return nil, fmt.Errorf("unexpected ProofOp.Type; got %v, want %v", pop.Type, ProofOpMultiStore)
-	}
-
-	// XXX: a bit strange as we'll discard this, but it works
-	var op MultiStoreProofOp
-
-	err := cdc.UnmarshalBinaryBare(pop.Data, &op)
-	if err != nil {
-		return nil, fmt.Errorf("decoding ProofOp.Data into MultiStoreProofOp: %w", err)
-	}
-
-	return NewMultiStoreProofOp(pop.Key, op.Proof), nil
-}
-
-// ProofOp return a merkle proof operation from a given multi-store proof
-// operation.
-func (op MultiStoreProofOp) ProofOp() merkle.ProofOp {
-	bz := cdc.MustMarshalBinaryBare(op)
-	return merkle.ProofOp{
-		Type: ProofOpMultiStore,
-		Key:  op.key,
-		Data: bz,
-	}
-}
-
-// String implements the Stringer interface for a mult-store proof operation.
-func (op MultiStoreProofOp) String() string {
-	return fmt.Sprintf("MultiStoreProofOp{%v}", op.GetKey())
-}
-
-// GetKey returns the key for a multi-store proof operation.
-func (op MultiStoreProofOp) GetKey() []byte {
-	return op.key
-}
-
-// Run executes a multi-store proof operation for a given value. It returns
-// the root hash if the value matches all the store's commitID's hash or an
-// error otherwise.
-func (op MultiStoreProofOp) Run(args [][]byte) ([][]byte, error) {
-	if len(args) != 1 {
-		return nil, errors.New("value size is not 1")
-	}
-
-	value := args[0]
-	root := op.Proof.ComputeRootHash()
-
-	for _, si := range op.Proof.StoreInfos {
-		if si.Name == string(op.key) {
-			if bytes.Equal(value, si.Core.CommitID.Hash) {
-				return [][]byte{root}, nil
-			}
-
-			return nil, fmt.Errorf("hash mismatch for substore %v: %X vs %X", si.Name, si.Core.CommitID.Hash, value)
-		}
-	}
-
-	return nil, fmt.Errorf("key %v not found in multistore proof", op.key)
-}
-
-//-----------------------------------------------------------------------------
-
 // XXX: This should be managed by the rootMultiStore which may want to register
 // more proof ops?
 func DefaultProofRuntime() (prt *merkle.ProofRuntime) {
 	prt = merkle.NewProofRuntime()
 	prt.RegisterOpDecoder(merkle.ProofOpSimpleValue, merkle.SimpleValueOpDecoder)
 	prt.RegisterOpDecoder(iavl.ProofOpIAVLValue, iavl.ValueOpDecoder)
 	prt.RegisterOpDecoder(iavl.ProofOpIAVLAbsence, iavl.AbsenceOpDecoder)
-	prt.RegisterOpDecoder(ProofOpMultiStore, MultiStoreProofOpDecoder)
```

Comment:
If you can drop this one (:+1:) you an also add another proof op that returns ics23 format, no?